File: 739-daily-temperatures/daily-temperatures.java
import java.util.Stack;

class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int result[] = new int[temperatures.length];
        Stack<Integer> s = new Stack<Integer>();
        Stack<Integer> si = new Stack<Integer>();
        for(int i = 0; i < temperatures.length; i++) result[i] = 0;

        for(int i = 0; i < temperatures.length; i++){
            while(!s.empty() && temperatures[i]>s.peek()){
                result[si.peek()] = i - si.peek();
                s.pop();
                si.pop();
            }
            s.push(temperatures[i]);
            si.push(i);
        }
        return result;
    }
}

File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        result = [0] * len(temperatures)
        stack = []
        for i, j in enumerate(temperatures):
            while stack and j > stack[-1][1]:
                k, l = stack.pop()
                result[k] = i - k
            stack.append((i, j))
        return result